src/Shared/Microsoft.DotNet.Scaffolding.Shared/CodeModifier/DocumentBuilder.cs

```diff
@@ -355,327 +431,219 @@ private BaseMethodDeclarationSyntax AddParameters(BaseMethodDeclarationSyntax me
+            }
+
+            if (updatedLambda == null)
+            {
+                return existingLambda;
             }
-            return modifiedBlockSyntaxNode;
+
+            return updatedLambda;
         }
 
-        internal ExpressionStatementSyntax AddExpressionInLambdaBlock(
-            ExpressionStatementSyntax expression,
-            string insertAfterBlock,
-            string codeSnippet,
-            IDictionary<string, string> parameterValues)
+        private static LambdaExpressionSyntax UpdateLambdaBlock(LambdaExpressionSyntax existingLambda, IEnumerable<SyntaxNode> lambdaChildren, CodeSnippet change)
         {
-            BlockSyntax blockToEdit;
-            if (!string.IsNullOrEmpty(codeSnippet))
+            if (ProjectModifierHelper.StatementExists(existingLambda.Block, change.Block))
             {
-                if (!string.IsNullOrEmpty(insertAfterBlock))
-                {
-                    string insertAfterFormattedBlock = ProjectModifierHelper.FormatCodeBlock(insertAfterBlock, parameterValues);
-                    blockToEdit = expression.DescendantNodes().FirstOrDefault(node =>
-                                    node is BlockSyntax &&
-                                    node.ToString().Trim(ProjectModifierHelper.CodeSnippetTrimChars).Contains(insertAfterBlock)) as BlockSyntax;
-                }
-                else
-                {
-                    blockToEdit = expression.DescendantNodes()
-                        .Where(node => node is BlockSyntax)
-                        .FirstOrDefault() as BlockSyntax;
-                }
+                return existingLambda;
+            }
 
-                if (blockToEdit != null)
-                {
-                    var sampleStatement = blockToEdit.Statements.FirstOrDefault();
-                    var innerTrailingTrivia = sampleStatement?.GetTrailingTrivia() ?? new SyntaxTriviaList();
-                    var innerLeadingTrivia = sampleStatement?.GetLeadingTrivia() ?? new SyntaxTriviaList();
+            var updatedBlock = AddStatement(existingLambda.Block, change);
+            if (existingLambda.WithBlock(updatedBlock as BlockSyntax) is LambdaExpressionSyntax updatedLambda)
+            {
+                return updatedLambda;
+            }
 
-                    if (!innerTrailingTrivia.Contains(SemiColonTrivia))
-                    {
-                        innerTrailingTrivia = innerTrailingTrivia.Insert(0, SemiColonTrivia);
-                    }
+            return existingLambda;
+        }
 
-                    StatementSyntax innerStatement = SyntaxFactory.ParseStatement(codeSnippet)
-                        .WithAdditionalAnnotations(Formatter.Annotation)
-                        .WithLeadingTrivia(innerLeadingTrivia)
-                        .WithTrailingTrivia(innerTrailingTrivia);
+        internal static SyntaxNode AddLambdaToParent(SyntaxNode parent, IEnumerable<SyntaxNode> children, CodeSnippet change)
```

Comment:
need comments to explain this method step by step. some of those if statements have a lot going on.